File: server/server-core/src/main/java/com/simple/castle/server/physic/ActiveObject.java
package com.simple.castle.server.physic;

import com.badlogic.gdx.physics.bullet.collision.btCollisionObject;
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody;

abstract class ActiveObject extends PhysicObject {
    ActiveObject(final btRigidBody.btRigidBodyConstructionInfo constructionInfo) {
        super(constructionInfo);
        this.body.setCollisionFlags(this.body.getCollisionFlags() | btCollisionObject.CollisionFlags.CF_CUSTOM_MATERIAL_CALLBACK);
        this.body.setContactCallbackFlag(Flags.OBJECT_FLAG);
        this.body.setContactCallbackFilter(Flags.GROUND_FLAG);
    }
}

File: server/server-core/src/main/java/com/simple/castle/server/kt/physic/ActiveObject.kt
package com.simple.castle.server.kt.physic

import com.badlogic.gdx.physics.bullet.collision.btCollisionObject
import com.badlogic.gdx.physics.bullet.dynamics.btRigidBody

class ActiveObject(constructionInfo: btRigidBody.btRigidBodyConstructionInfo) : PhysicObject(constructionInfo) {
    init {
        body.collisionFlags = body.collisionFlags or btCollisionObject.CollisionFlags.CF_CUSTOM_MATERIAL_CALLBACK
        body.contactCallbackFlag = Flags.OBJECT_FLAG.toInt()
        body.contactCallbackFilter = Flags.GROUND_FLAG.toInt()
    }
}